### ISSUE TITLE
submit a job , and the job  has not been ran。

### ISSUE THREAD

**Issue by poetryben88** [microsoft/pai#5659]
openpai/openpai-runtime container   on pai-work node     error log：

root@pai-worker1:/usr/local# docker logs 6e354aae0a40
+ CHILD_PROCESS=UNKNOWN
+ trap exit_handler EXIT
+ PAI_WORK_DIR=/usr/local/pai
+ PAI_CONFIG_DIR=/usr/local/pai-config
+ PAI_INIT_DIR=/usr/local/pai/init.d
+ PAI_RUNTIME_DIR=/usr/local/pai/runtime.d
+ PAI_LOG_DIR=/usr/local/pai/logs/2e4061a5-244c-45af-8ede-b4b9f6604774
+ PAI_SECRET_DIR=/usr/local/pai/secrets
+ PAI_USER_EXTENSION_SECRET_DIR=/usr/local/pai/user-extension-secrets
+ PAI_TOKEN_SECRET_DIR=/usr/local/pai/token-secrets
+ chmod a+rw /usr/local/pai/logs/2e4061a5-244c-45af-8ede-b4b9f6604774
+ find /usr/local/pai/logs/2e4061a5-244c-45af-8ede-b4b9f6604774 -maxdepth 1 -type f '!' -name init.log
+ LOG_FILES=
+ '[[' '!' -z  ]]
+ find /usr/local/pai -maxdepth 1 -mindepth 1 '!' -name logs -exec rm -rf '{}' ';'
rm: can't remove '/usr/local/pai/user-extension-secrets/..data': Read-only file system
rm: can't remove '/usr/local/pai/user-extension-secrets/userExtensionSecrets.yaml': Read-only file system
rm: can't remove '/usr/local/pai/user-extension-secrets/..2021_11_18_08_54_30.144092891/userExtensionSecrets.yaml': Read-only file system
+ mv ./__init__.py ./common ./init ./init.d ./package_cache ./plugins ./requirements.txt ./runtime ./runtime.d /usr/local/pai
+ cd /usr/local/pai
+ '[[' true '=' true ]]
+ CHILD_PROCESS=FRAMEWORK_BARRIER
+ echo 'frameworkbarrier start'
frameworkbarrier start
+ '[[' -f /var/run/secrets/kubernetes.io/serviceaccount/token ]]
+ unset KUBE_APISERVER_ADDRESS
+ /usr/local/pai/init.d/frameworkbarrier
+ tee /usr/local/pai/logs/2e4061a5-244c-45af-8ede-b4b9f6604774/barrier.log
I1118 08:54:33.933533      14 barrier.go:211] Initializing frameworkbarrier
I1118 08:54:33.933705      14 barrier.go:214] With Config:
kubeApiServerAddress: ""
kubeConfigFilePath: ""
frameworkNamespace: default
frameworkName: 3a3cada7aa94eff914b372e17e17c8e1
barrierCheckIntervalSec: 10
barrierCheckTimeoutSec: 600
W1118 08:54:33.933736      14 client_config.go:549] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I1118 08:54:33.934851      14 barrier.go:227] Running frameworkbarrier
W1118 08:54:33.945092      14 barrier.go:253] Failed to get Framework object from ApiServer: Unauthorized
W1118 08:54:43.947234      14 barrier.go:253] Failed to get Framework object from ApiServer: Unauthorized
W1118 08:54:53.947552      14 barrier.go:253] Failed to get Framework object from ApiServer: Unauthorized
W1118 08:55:03.947109      14 barrier.go:253] Failed to get Framework object from ApiServer: Unauthorized
W1118 08:55:13.947432      14 barrier.go:253] Failed to get Framework object from ApiServer: Unauthorized

webport log
![image](https://user-images.githubusercontent.com/15098245/142384730-a4e8754f-d3d4-4d6d-a9b6-9a50a65ecff9.png)

![image](https://user-images.githubusercontent.com/15098245/142384755-acec963f-3c29-4c38-b654-27078dd6ab7a.png)

**Comment by hzy46** [comment]
The cause should be this:

```
W1118 08:54:33.945092 14 barrier.go:253] Failed to get Framework object from ApiServer: Unauthorized
W1118 08:54:43.947234 14 barrier.go:253] Failed to get Framework object from ApiServer: Unauthorized
W1118 08:54:53.947552 14 barrier.go:253] Failed to get Framework object from ApiServer: Unauthorized
W1118 08:55:03.947109 14 barrier.go:253] Failed to get Framework object from ApiServer: Unauthorized
W1118 08:55:13.947432 14 barrier.go:253] Failed to get Framework object from ApiServer: Unauthorized
```

In your pod, K8s api server is not authorized. Could you please provide your pod definition? Maybe there's something wrong with the service account.

**Comment by poetryben88** [comment]
@hzy46    thanks for your  response。

Maybe the worker has no ROLES？

![image](https://user-images.githubusercontent.com/15098245/144162155-cf8623ce-186b-4a93-b233-b4703b2907d7.png)
![image](https://user-images.githubusercontent.com/15098245/144162221-28be0c83-ad21-4ca6-9006-b358831968fe.png)

**Comment by hzy46** [comment]
No, I mean service account. The pod should use the service account `runtime-account` to get the framework objects.

Please show me the results of the following command:

```
kubectl get sa runtime-account -o yaml
kubectl get clusterrole runtime-framework-role -o yaml
kubectl get clusterrolebinding runtime-framework-role-binding -o yaml
```
Reference: https://github.com/microsoft/pai/blob/7cf1ae351dd15d7228d2675212c1036c0aea8745/src/openpai-runtime/deploy/rbac.yaml

**Comment by poetryben88** [comment]
@hzy46 

Hi, Here is my results. I can not find any errors.
```
root@pai-master:/home/xubaishuai# kubectl get sa runtime-account -o yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","kind":"ServiceAccount","metadata":{"annotations":{},"name":"runtime-account","namespace":"default"}}
  creationTimestamp: "2021-10-11T09:32:15Z"
  name: runtime-account
  namespace: default
  resourceVersion: "272755"
  selfLink: /api/v1/namespaces/default/serviceaccounts/runtime-account
  uid: e5d6431e-982e-4a50-a128-50a6c095add3
secrets:
- name: runtime-account-token-bchmp
```
```
root@pai-master:/home/xubaishuai# kubectl get clusterrole runtime-framework-role -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"rbac.authorization.k8s.io/v1","kind":"ClusterRole","metadata":{"annotations":{},"name":"runtime-framework-role"},"rules":[{"apiGroups":["frameworkcontroller.microsoft.com"],"resources":["frameworks"],"verbs":["get","watch","list"]}]}
  creationTimestamp: "2021-10-11T09:32:15Z"
  name: runtime-framework-role
  resourceVersion: "272753"
  selfLink: /apis/rbac.authorization.k8s.io/v1/clusterroles/runtime-framework-role
  uid: 5d245afd-3a5f-45c2-b96f-da56d3443dfc
rules:
- apiGroups:
  - frameworkcontroller.microsoft.com
  resources:
  - frameworks
  verbs:
  - get
  - watch
  - list
```
```
root@pai-master:/home/xubaishuai# kubectl get clusterrolebinding runtime-framework-role-binding -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"rbac.authorization.k8s.io/v1","kind":"ClusterRoleBinding","metadata":{"annotations":{},"name":"runtime-framework-role-binding"},"roleRef":{"apiGroup":"rbac.authorization.k8s.io","kind":"ClusterRole","name":"runtime-framework-role"},"subjects":[{"kind":"ServiceAccount","name":"runtime-account","namespace":"default"}]}
  creationTimestamp: "2021-10-11T09:32:15Z"
  name: runtime-framework-role-binding
  resourceVersion: "272757"
  selfLink: /apis/rbac.authorization.k8s.io/v1/clusterrolebindings/runtime-framework-role-binding
  uid: 1d9301ff-5493-431b-8b1b-64e96ee43d6e
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: runtime-framework-role
subjects:
- kind: ServiceAccount
  name: runtime-account
  namespace: default
```
